app/experimenter/nimbus-ui/src/components/AppLayoutSidebarLocked/index.tsx

```diff
@@ -91,18 +76,141 @@ export const AppLayoutSidebarLocked = ({
     analysis?.metadata?.metrics || {},
   );
 
+  const getNestedSidebarItems = (
+    metrics: { [metric: string]: string },
+    title: string,
+  ) => {
+    const sidebarItems = Object.keys(metrics).map((sidebarKey) => {
+      if (analysisRequired) {
+        return (
+          <li className="ml-4 mb-2" key={metrics[sidebarKey]}>
+            <a
+              href={`#${sidebarKey}`}
+              className="inherit-color font-weight-normal"
+            >
+              {metrics[sidebarKey]}
+            </a>
+          </li>
+        );
+      }
+      return (
+        <LinkNav
+          key={metrics[sidebarKey]}
+          route={`${slug}/results#${sidebarKey}`}
+          storiesOf="pages/Results"
+          textColor="inherit-color"
+          className="font-weight-normal ml-4 mb-2"
+        >
+          {metrics[sidebarKey]}
+        </LinkNav>
+      );
+    });
+
+    // Semantically speaking, nested items should be wrapped in another `<ul>`
+    // but Scrollspy doesn't work as of 3.4.3 so all links are top-level under
+    // a single `<ul>`
+    sidebarItems.unshift(
+      <li key={title} className="mb-2 ml-3">
+        {title}
+      </li>,
+    );
+    return sidebarItems;
+  };
+
   const sidebarKeys = [
     "monitoring",
     "overview",
-    "results-summary",
-    "primary-metrics",
+    "results_summary",
+    "primary_metrics",
```

Comment:
Ah @emtwo I could revert these if preferred, but I don't mind leaving them since the data is `snake_case` and could match better this way. I created a helper function for these when I tried DRY'ing all this up that was useful with all of these as snake... until I discovered how wacky Scrollspy is.

Reply:
this wfm